frontend/src/pages/list/SalesReport.tsx

```diff
@@ -0,0 +1,273 @@
+  { field: "revenue", header: "Revenue" },
+  { field: "cost", header: "Cost" },
+  { field: "profit", header: "Profit" },
+];
+
+const COLUMNS_TOP_BOOKS: TableColumnSales[] = [
+  { field: "book_title", header: "Book" },
+  { field: "book_id", header: "id" },
+  { field: "num_books_sold", header: "Number of Books Sold" },
+  { field: "book_revenue", header: "Book Revenue" },
+  { field: "total_cost_most_recent", header: "Book Cost - Most Recent" },
+  { field: "book_profit", header: "Book Profit" },
+];
+
+export default function SalesReport() {
+  const [totalsData, setTotalsData] = useState<salesReportTotalRow[]>([]);
+  const [dailyData, setDailyData] = useState<salesReportDailyRow[]>([]);
+  const [topBooksData, setTopBooksData] = useState<salesReportTopBooksRow[]>(
+    []
+  );
+  const [dates, setDates] = useState<any>(null);
+  const [isConfirmationPopupVisible, setIsConfirmationPopupVisible] =
+    useState(false);
+
+  const onAPIResponse = (response: GetSalesReportResp) => {
+    setTotalsData([
+      {
+        revenue: response.revenue,
+        cost: response.cost,
+        profit: response.profit,
+      },
+    ]);
+    setDailyData(response.daily_summary);
+    setTopBooksData(response.top_books);
+    showSuccess("Data Successfully Retrieved");
+  };
+
+  // Toast is used for showing success/error messages
+  const toast = useRef<Toast>(null);
+
+  const showSuccess = (message: string) => {
+    toast.current?.show({ severity: "success", summary: message });
+  };
+
+  const showFailure = (message: string) => {
+    toast.current?.show({
+      severity: "error",
+      summary: message,
+    });
+  };
+
+  const formatCurrency = (value: any) => {
+    return value.toLocaleString("en-US", {
+      style: "currency",
+      currency: "USD",
+    });
+  };
+
+  const priceBodyTemplateRevenue = (rowData: any) => {
+    return formatCurrency(rowData.revenue);
+  };
+  const priceBodyTemplateCost = (rowData: any) => {
+    return formatCurrency(rowData.cost);
+  };
+  const priceBodyTemplateProfit = (rowData: any) => {
+    return formatCurrency(rowData.profit);
+  };
+
+  const priceBodyTemplateBookRevenue = (rowData: any) => {
+    return formatCurrency(rowData.book_revenue);
+  };
+  const priceBodyTemplateBookCost = (rowData: any) => {
+    return formatCurrency(rowData.total_cost_most_recent);
+  };
+  const priceBodyTemplateBookProfit = (rowData: any) => {
+    return formatCurrency(rowData.book_profit);
+  };
+
+  const onSubmit = (): void => {
+    try {
+      if (dates != null) {
+        if (moment(dates[1]).format("YYYY-MM-DD") !== "Invalid date") {
+          //console.log(moment(new Date()).format("YYYY-MM-DD"));
+          // console.log(moment(dates[0]).format("YYYY-MM-DD"));
+          // console.log(moment(dates[1]).format("YYYY-MM-DD"));
+          logger.debug("Sales Report Requested");
+          SALES_REPORT_API.getSalesReport({
+            start: moment(dates[0]).format("YYYY-MM-DD"),
```

Comment:
Can you explain what this does? I added a util function (util/DateOperations), I'm just not sure what "moment" does